### PR TITLE
update: upgrading kotlin version and kotlin compiler extension version

### PR DESCRIPTION
**Pull request by Ibnyahyah**
### **User description**
Story: https://app.shortcut.com/smileid/story/xxx

## Summary

A few sentences/bullet points about the changes

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

___

### **PR Type**
Enhancement

___

### **Description**
- Upgrade Kotlin and related dependencies

- Update Gradle wrapper to version 8.10.2

- Add VSCode Java configuration settings

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>settings.json</strong><dd><code>Add VSCode Java configuration settings</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.vscode/settings.json

<li>Added new VSCode settings file with Java configuration<br> <li> Set Java build configuration update mode to "interactive"

</details>

  </td>
  <td><a href="https://github.com/smileidentity/flutter/pull/159/files#diff-a5de3e5871ffcc383a2294845bd3df25d3eeff6c29ad46e3a396577c413bf357">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Dependencies</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>gradle-wrapper.properties</strong><dd><code>Upgrade Gradle wrapper to version 8.10.2</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

android/gradle/wrapper/gradle-wrapper.properties

<li>Updated Gradle distribution URL from 8.9-bin.zip to 8.10.2-all.zip<br> <li> Upgraded to newer Gradle version with full distribution package

</details>

  </td>
  <td><a href="https://github.com/smileidentity/flutter/pull/159/files#diff-4c5ffadbdf6016477a1c3768e0b7381cc298bd558b1784871f407cbcb0efacce">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about PR-Agent usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "java.configuration.updateBuildConfiguration": "interactive"
```
Comment (jumaallan) [inline .vscode/settings.json:2]: remove this

File: android/build.gradle [hunk 2]
```diff
@@ -1,7 +1,7 @@
-def kotlinVersion = findProperty('kotlinVersion') ?: '2.1.0'
+def kotlinVersion = findProperty('kotlinVersion') ?: '2.1.20' // Ensure using Kotlin 2.1.20
 //def smileVersion = findProperty('smileVersion') ?: '10.6.1-SNAPSHOT' // Uncomment this line to use the latest snapshot version also uncomment the snapshot repository
 def smileVersion = findProperty('smileVersion') ?: '10.6.0'
-def kotlinCompilerExtension = findProperty('kotlinCompilerExtensionVersion') ?: '1.5.14'
+def kotlinCompilerExtension = findProperty('kotlinCompilerExtensionVersion') ?: '2.0.0' // Update to match Kotlin version
```
Comment (jumaallan) [inline android/build.gradle:4]: See this - https://developer.android.com/jetpack/androidx/releases/compose-kotlin

this is no longer needed after kotlin 2.0

**Review by jumaallan — CHANGES_REQUESTED** [review]
Requested some changes
